
#KIDNEY RENAL CLEAR-CELL-CARCINOMA EXPRESSION ANALYSE

## Table of Contents

- [LOADING DATA](#loading-data)
	- [Explore phenotypic data](#exploring-phenotypic-data)
	- [Explore feature data](#exploring-feature-data)
	- [DGE object](#dge-object)
- [QUALITY AND NORMALIZATION](#quality-&-normalization)
	- [Paired samples](#paired-samples-removing)
	- [Library size filtering](#library-size-filtering)
	- [Genes Filtering: Distribution of expression levels among genes](#genes-filtering-distribution-of-expression-levels-among-genes)
		- [Expression levels comparison between all sampes](#expression-levels-comparison-between-all-samples)
	  - [Expression levels comparison between normal and tumor](#expression-levels-comparison-between-normal-and-tumor)
		- [Distribution of expression levels among genes](#distribution-of-expression-levels-among-genes)
	- [MA-plots](#ma-plots)
	- [BATCH IDENTIFICATION](#batch-identification)
	  - [TSS batch effect](#tss-batch-effect)
	  - [PLATE batch effect](#plate-batch-effect)
	  - [PORTION-ANALYTE batch effect](#portion-analyte-batch-effect)
	- [Surrogate variable analysis](#surrogate-variable-analysis)


```
library(knitr)     ## required for "knitting" from Rmd to md
library(markdown)  ## required for processing from md to HTML
knit2html("~/Dropbox/Home/Documentos/MASTER/IEO/KIDNEY/project.Rmd")  ## process Rmd to HTML
browseURL("projectTemplate.html") ## open the resulting HTML file from R
```


##LOADING DATA

```{r}
library(SummarizedExperiment)
```

We load the raw RNA-seq data counts set of Kidney renal clear-cell-carcinoma.
```{r}
se <- readRDS(file.path("~/Documents/1.Master/Github/KIDNEY/seKIRC.rds"))
se
```

###Exploring phenotypic data

<We take a look to the column data, which include the phenotipic information of all samples,of the sumarized experiment object.
```{r}
dim(colData(se))
colData(se)[1:5, 1:5]
```
We also need to observe to the metadata information content of these data.
```{r}
mcols(colData(se), use.names=TRUE)
```
As we can see there are three different columns on the metadata result. The first column is the name of all the variables; the second is the labelDescription, which is the definition of each variable; and the last one corresponding to the Common Data Element Identifier (CDEID).

###Exploring feature data
We look through the data rows, which are the feature elements, to see the genes information of our data set.
```{r}
rowRanges(se)
```


###DGE object
We create a DGEList' object to hold the dataset to be analysed in a better and more comprehensive way.
```{r}
library(edgeR)
dge <- DGEList(counts=assays(se)$counts, genes=mcols(se))
```
Moreover, we calculate $\log_2$ CPM values of expression and we save them in an assay element.
```{r}
assays(se)$logCPM <- cpm(dge, log=TRUE, prior.count=0.5)
```


#QUALITY & NORMALIZATION
##Paired samples removing
```{r}
dge.filt <- dge
```

##Library size filtering
At this point, we need to determine

```{r}
library(ggplot2)

libsize <- data.frame(libsize = dge$sample$lib.size/1e6, type = se$type)

summary(libsize)


ggplot(libsize) +
  geom_density(aes(x=libsize, fill=type), alpha=0.5) +
  ylab("density\n") + xlab("\nMillions of Reads") +
  theme_bw()

ggplot(libsize) +
  geom_histogram(
    aes(x=libsize, fill=type, y = (..count..)/sum(..count..)), alpha=0.5, binwidth=3
  ) + xlab("\nMillions of Reads") + ylab("% of Samples\n") + theme_bw()

dge.filt <- dge.filt[, dge.filt$sample$lib.size/1e6 > 50 & dge.filt$sample$lib.size/1e6 < 70]


final.samples <- rownames(dge.filt$samples)

se.filt <-se[,colnames(se) %in% final.samples]

summary(se.filt$samples)


```

POSAR EXPLICACIÓ: EL SEGÜENT, ÉS DESPRÉS DE FILTAR
```{r}

summary(se.filt$type)

ggplot(libsize) +
  geom_histogram(
    aes(x=libsize, fill=type, y = (..count..)/sum(..count..)), alpha=0.5, binwidth=3
  ) + xlab("\nMillions of Reads") + ylab("% of Samples\n") + theme_bw() + xlim(c(25, 71))

ggplot(libsize) +
  geom_density(aes(x=libsize, fill=type), alpha=0.5) +
  ylab("density\n") + xlab("\nMillions of Reads") +
  theme_bw() + xlim(c(25, 71))

```


###Genes filtering: Distribution of expression levels
```{r}
library(geneplotter)
par(mfrow=c(1,2), mar=c(4,5,1,1))
```



####Expression levels comparison between all samples
```{r}
multidensity(as.list(as.data.frame(assays(se.filt)$logCPM)), xlab = "log2 CPM", legend = NULL, main = "All samples", cex.axis = 1.2, cex.lab = 1.5, las = 1)
```

####Expression levels comparison between normal and tumor
```{r}
normalCPM <- se.filt[,se.filt$type == "normal"]
tumorCPM <- se.filt[,se.filt$type == "tumor"]
multidensity(as.list(as.data.frame(assays(normalCPM)$logCPM)), xlab = "log2 CPM", legend = NULL, main = "Normal", cex.axis = 1.2, cex.lab = 1.5, las = 1)
multidensity(as.list(as.data.frame(assays(tumorCPM)$logCPM)), xlab = "log2 CPM", legend = NULL, main = "Tumor", cex.axis = 1.2, cex.lab = 1.5, las = 1)
```

####Distribution of expression levels among genes
```{r}
genemean <- data.frame(Means=rowMeans(assays(se.filt)$logCPM[,-1]))

ggplot(genemean) +
  geom_bar(aes(x=Means), binwidth = 1, fill="white", color="black") +
  theme_bw() + xlab("\nlog2CPM") + ylab("Count\n") +
  geom_vline(xintercept=mean(genemean$Means), color="red")
```

EXPLICACIÓ PERQUÈ FEM EL FILTRE SEGÜENT
```{r}
avgexp <- rowMeans(assays(se.filt)$logCPM)
mask <- avgexp > mean(genemean$Means)

dim(se.filt) # Before filtering SE
se.filt.genes <- se.filt[mask, ]
dim(se.filt.genes) # After filtering SE

dim(dge.filt) # Before filtering DGE
dge.filt <- dge.filt[mask, ]
dim(dge.filt) # After filtering DGE

```

```{r}
dgenorm <- calcNormFactors(dge.filt)
assays(se.filt.genes)$logCPMnorm <- cpm(dgenorm, log=TRUE, prior.count=0.5)
```


```{r libsizes, echo=FALSE, out.width="600px", fig.cap="Figure S1: Library sizes in increasing order."}
ord <- order(dge.filt$sample$lib.size/1e6)
barplot(dge.filt$sample$lib.size[ord]/1e6, las=1, ylab="Millions of reads",
        xlab="Samples", col=c("blue", "red")[(se.filt.genes$type[ord] == "tumor") + 1])
legend("topleft", c("tumor", "normal"), fill=c("red", "blue"), inset=0.01)
```


###MA-plots

We look at the MA-plots of the normal samples to see if there are systematic biases in gene expression levels in any of the sampes. We expect the slope to be around zero.


```{r maPlotsNormal, fig.height=28, fig.width=10, dpi=100, echo=FALSE, fig.cap="Figure S4: MA-plots of the normal samples."}
par(mfrow=c(9, 4 ), mar=c(4, 4, 3, 1))
setmp <- se.filt.genes[, se.filt.genes$type == "normal"]
dgetmp <- dge.filt[, se.filt.genes$type == "normal"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPMnorm)
  M <- assays(setmp)$logCPMnorm[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

As we can see, we don't see any samples with major biases in gene expression.

Now, we examine the tumor samples:

```{r maPlotsTumor, fig.height=110, fig.width=10, dpi=100, echo=FALSE, fig.cap="Figure S4: MA-plots of the tumor samples."}
par(mfrow=c(45, 4 ), mar=c(2, 2, 1, 1))
setmp <- se.filt.genes[, se.filt.genes$type == "tumor"]
dgetmp <- dge.filt[, se.filt.genes$type == "tumor"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPMnorm)
  M <- assays(setmp)$logCPMnorm[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

<br/>   

###Batch identification
```{r}
tss <- substr(colnames(se.filt.genes), 6, 7)
table(tss)
center <- substr(colnames(se.filt.genes), 27, 28)
table(center)
plate <- substr(colnames(se.filt.genes), 22, 25)
table(plate)
portionanalyte <- substr(colnames(se.filt.genes), 18, 20)
table(portionanalyte)
samplevial <- substr(colnames(se.filt.genes), 14, 16)
table(samplevial)
#dge.filt
#se.filt.genes

colnames(se.filt.genes)[portionanalyte == "02R"]
colnames(se.filt.genes)[portionanalyte == "11R"]

```

####TSS batch effect

```{r}
table(data.frame(TYPE=se.filt.genes$type, TSS=tss))
```
```{r sampleClusteringTSS, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S6: Hierarchical clustering of the samples."}
#logCPM <- cpm(dge.filt, log=TRUE, prior.count=3)
d <- as.dist(1-cor(assays(se.filt.genes)$logCPMnorm, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(tss))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(se.filt.genes)
outcome <- paste(substr(colnames(se.filt.genes), 9, 12), as.character(se.filt.genes$type), sep="-")
names(outcome) <- colnames(se.filt.genes)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("TSS", sort(unique(batch)), levels(factor(tss))), fill=sort(unique(batch)))

```

```{r mdsTSS, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S7: Multidimensional scaling plot of the samples."}
plotMDS(dge.filt, labels=outcome, col=batch)
legend("bottomleft", paste("TSS", sort(unique(batch)), levels(factor(tss))),
       fill=sort(unique(batch)), inset=0.05)
```



####PLATE batch effect

```{r}
table(data.frame(TYPE=se.filt.genes$type, PLATE=plate))
```

```{r sampleClusteringPLATE, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S6: Hierarchical clustering of the samples."}
#logCPM <- cpm(dge.filt, log=TRUE, prior.count=3)
d <- as.dist(1-cor(assays(se.filt.genes)$logCPMnorm, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(plate))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(se.filt.genes)
outcome <- paste(substr(colnames(se.filt.genes), 9, 12), as.character(se.filt.genes$type), sep="-")
names(outcome) <- colnames(se.filt.genes)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("PLATE", sort(unique(batch)), levels(factor(plate))), fill=sort(unique(batch)))

```

```{r mdsPLATE, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S7: Multidimensional scaling plot of the samples."}
plotMDS(dge.filt, labels=outcome, col=batch)
legend("bottomleft", paste("plate", sort(unique(batch)), levels(factor(plate))),
       fill=sort(unique(batch)), inset=0.05)
```


####PORTION-ANALYTE batch effect

```{r}
table(data.frame(TYPE=se.filt.genes$type, P.analyte = portionanalyte))
```
```{r sampleClusteringPortionanalyte, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S6: Hierarchical clustering of the samples."}
#logCPM <- cpm(dge.filt, log=TRUE, prior.count=3)
d <- as.dist(1-cor(assays(se.filt.genes)$logCPMnorm, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(portionanalyte))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(se.filt.genes)
outcome <- paste(substr(colnames(se.filt.genes), 9, 12), as.character(se.filt.genes$type), sep="-")
names(outcome) <- colnames(se.filt.genes)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("Portionanalyte", sort(unique(batch)), levels(factor(portionanalyte))), fill=sort(unique(batch)))

```

```{r mdsPortionanalyte, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S7: Multidimensional scaling plot of the samples."}
plotMDS(dge.filt, labels=outcome, col=batch)
legend("bottomleft", paste("Portionanalyte", sort(unique(batch)), levels(factor(portionanalyte))),
       fill=sort(unique(batch)), inset=0.05)

```



### Filtering out samples

4770, 4698, 6087, 4983


<br/>   

###Surrogate variable analysis

```{r}
library(sva)
mod <- model.matrix(~ se.filt.genes$type, colData(se.filt.genes))
modzero <- model.matrix(~ 1, colData(se.filt.genes))
pv <- f.pvalue(assays(se.filt.genes)$logCPMnorm, mod, modzero)
sum(p.adjust(pv, method="fdr") < 0.01)
hist(pv,main="",las=1)

sv <- sva(assays(se.filt.genes)$logCPMnorm, mod, modzero)
sv$n
modsv <- cbind(mod, sv$sv)
modzerosv <- cbind(modzero, sv$sv)
pvsv <- f.pvalue(assays(se.filt.genes)$logCPMnorm, modsv, modzerosv)
sum(p.adjust(pvsv, method="fdr") < 0.01)
hist(pvsv,main="",las=1)
