
#KIDNEY RENAL CLEAR-CELL-CARCINOMA EXPRESSION ANALYSE

## Table of Contents

- [LOADING DATA](#loading-data)
	- [Explore phenotypic data](#exploring-phenotypic-data)
	- [Explore feature data](#exploring-feature-data)
	- [DGE object](#dge-object)
- [QUALITY AND NORMALIZATION](#quality-&-normalization)
	- [Paired samples](#paired-samples-removing)
	- [Library size filtering](#library-size-filtering)
	- [Genes Filtering: Distribution of expression levels among genes](#genes-filtering-distribution-of-expression-levels-among-genes)
		- [Expression levels comparison between all sampes](#expression-levels-comparison-between-all-samples)
	  - [Expression levels comparison between normal and tumor](#expression-levels-comparison-between-normal-and-tumor)
		- [Distribution of expression levels among genes](#distribution-of-expression-levels-among-genes)
	- [MA-plots](#ma-plots)
	- [Batch identification](#batch-identification)
	  - [TSS batch effect](#tss-batch-effect)
	  - [PLATE batch effect](#plate-batch-effect)
	  - [PORTION-ANALYTE batch effect](#portion-analyte-batch-effect)
	- [Surrogate variable analysis](#surrogate-variable-analysis)
- [CONCLUSIONS](#conclusions)


```
library(knitr)     ## required for "knitting" from Rmd to md
library(markdown)  ## required for processing from md to HTML
knit2html("~/Dropbox/Home/Documentos/MASTER/IEO/KIDNEY/project.Rmd")  ## process Rmd to HTML
browseURL("projectTemplate.html") ## open the resulting HTML file from R
```


##LOADING DATA

```{r}
library(SummarizedExperiment)
```


We load the raw RNA-seq data counts set of Kidney renal clear-cell-carcinoma.
```{r}
se <- readRDS(file.path("~/Documentos/MASTER/IEO/KIDNEY/seKIRC.rds"))

se
```

###Exploring phenotypic data

<We take a look to the column data, which include the phenotipic information of all samples,of the sumarized experiment object.
```{r}
dim(colData(se))
colData(se)[1:5, 1:5]
```
We also need to observe to the metadata information content of these data.
```{r}
mcols(colData(se), use.names=TRUE)
```
As we can see there are three different columns on the metadata result. The first column is the name of all the variables; the second is the labelDescription, which is the definition of each variable; and the last one corresponding to the Common Data Element Identifier (CDEID).

###Exploring feature data
We look through the data rows, which are the feature elements, to see the genes information of our data set.
```{r}
rowRanges(se)
```


###DGE object
We create a DGEList' object to hold the dataset to be analysed in a better and more comprehensive way.
```{r}
library(edgeR)
dge <- DGEList(counts=assays(se)$counts, genes=mcols(se))
```
Moreover, we calculate $\log_2$ CPM values of expression and we save them in an assay element.
```{r}
assays(se)$logCPM <- cpm(dge, log=TRUE, prior.count=0.5)
```


#QUALITY & NORMALIZATION
##Paired samples removing
```{r}
dge.filt <- dge
```

##Library size filtering
At this point, we need to determine if the library sizes of tumor samples and normal samples are similar or not.

```{r}
library(ggplot2)

libsize <- data.frame(libsize = dge$sample$lib.size/1e6, type = se$type)

summary(libsize)


ggplot(libsize) +
  geom_density(aes(x=libsize, fill=type), alpha=0.5) +
  ylab("density\n") + xlab("\nMillions of Reads") +
  theme_bw()
```

This past figure shows the density distribution on sequencing depth (milions of reads).

```{r}
ggplot(libsize) +
  geom_histogram(
    aes(x=libsize, fill=type, y = (..count..)/sum(..count..)), alpha=0.5, binwidth=3
  ) + xlab("\nMillions of Reads") + ylab("% of Samples\n") + theme_bw()
```

In this other figure, we can see now the proporcion of samples in each sequencing depth (Milions of Reads).

As we can see, both figures show that the normal and tumor samples are similar in terms of sequencing depth, except for some samples that show extreme values.

Moreover, we can see how both normal and tumor types, follow a normal distribution through the sequencing depth.

At this point, we can filter out those samples with a low sequencing depth (<50 Milions of reads) because they are less reliable. Furthermore, we also remove the samples with extreme values explained before (>70 Milions of Reads).
```{r}
dge.filt <- dge.filt[, dge.filt$sample$lib.size/1e6 > 50 & dge.filt$sample$lib.size/1e6 < 70]


final.samples <- rownames(dge.filt$samples)

se.filt <-se[,colnames(se) %in% final.samples]

```
These are the numbers of tumor and normal samples before filtering
```{r}
summary(se$type)
```

And now after filtering:
```{r}
summary(se.filt$type)
```

As we can see in the following figure, the distribution of normal and tumor samples is very similar:

```{r libsizes, echo=FALSE, out.width="600px", fig.cap="Figure S1: Library sizes in increasing order."}
ord <- order(dge.filt$sample$lib.size/1e6)
barplot(dge.filt$sample$lib.size[ord]/1e6, las=1, ylab="Millions of reads",
        xlab="Samples", col=c("blue", "red")[(se.filt$type[ord] == "tumor") + 1])
legend("topleft", c("tumor", "normal"), fill=c("red", "blue"), inset=0.01)
```

###Genes filtering: Distribution of expression levels
```{r}
library(geneplotter)
par(mfrow=c(1,2), mar=c(4,5,1,1))
```

####Expression levels comparison between all samples
In the next part, we take a look on the distribution of expression values per sample in terms of logarithmic CPM units.

```{r}
multidensity(as.list(as.data.frame(assays(se.filt)$logCPM)), xlab = "log2 CPM", legend = NULL, main = "All samples", cex.axis = 1.2, cex.lab = 1.5, las = 1)
```

We cannot observe significant diferences between samples in terms of expression levels.

As we have more than 200 samples, we now display the normal and tumor distribution separately.

####Expression levels comparison between normal and tumor
```{r}
normalCPM <- se.filt[,se.filt$type == "normal"]
tumorCPM <- se.filt[,se.filt$type == "tumor"]
multidensity(as.list(as.data.frame(assays(normalCPM)$logCPM)), xlab = "log2 CPM", legend = NULL, main = "Normal", cex.axis = 1.2, cex.lab = 1.5, las = 1)
multidensity(as.list(as.data.frame(assays(tumorCPM)$logCPM)), xlab = "log2 CPM", legend = NULL, main = "Tumor", cex.axis = 1.2, cex.lab = 1.5, las = 1)
```

####Distribution of expression levels among genes
Now, we want to filter out the genes with a low expression. To do that, we can plot the CPMs in logarithmic scale.
```{r}
genemean <- data.frame(Means=rowMeans(assays(se.filt)$logCPM[,-1]))

ggplot(genemean) +
  geom_bar(aes(x=Means), binwidth = 1, fill="white", color="black") +
  theme_bw() + xlab("\nlog2CPM") + ylab("Count\n") +
  geom_vline(xintercept=mean(genemean$Means), color="red")
```

We can see how there are two peaks of genes. The first one correspond to these genes with a very low expression and thus we decide to remove them. In other to do that, we used the mean as a threshold.
```{r}
avgexp <- rowMeans(assays(se.filt)$logCPM)
mask <- avgexp > mean(genemean$Means)
```

These are the numbers of genes before filtering:
```{r}
dim(se.filt) # Before filtering SE
dim(dge.filt) # Before filtering DGE
```

These are the numbers of samples genes after filtering:
```{r}
se.filt.genes <- se.filt[mask, ]
dim(se.filt.genes) # After filtering SE

dge.filt <- dge.filt[mask, ]
dim(dge.filt) # After filtering DGE

```

At this point, we can calculate the normaliation factors on the filtered expression data set.
```{r}
dgenorm <- calcNormFactors(dge.filt)
assays(se.filt.genes)$logCPMnorm <- cpm(dgenorm, log=TRUE, prior.count=0.5)
```

###MA-plots

We look at the MA-plots of the normal samples to see if there are systematic biases in gene expression levels in any of the sampes. We expect the slope to be around zero.


```{r maPlotsNormal, fig.height=28, fig.width=10, dpi=100, echo=FALSE, fig.cap="Figure S4: MA-plots of the normal samples."}
par(mfrow=c(9, 4 ), mar=c(4, 4, 3, 1))
setmp <- se.filt.genes[, se.filt.genes$type == "normal"]
dgetmp <- dge.filt[, se.filt.genes$type == "normal"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPMnorm)
  M <- assays(setmp)$logCPMnorm[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

As we can see, we don't see any samples with major biases in gene expression.

Now, we examine the tumor samples:

```{r maPlotsTumor, fig.height=110, fig.width=10, dpi=100, echo=FALSE, fig.cap="Figure S4: MA-plots of the tumor samples."}
par(mfrow=c(45, 4 ), mar=c(2, 2, 1, 1))
setmp <- se.filt.genes[, se.filt.genes$type == "tumor"]
dgetmp <- dge.filt[, se.filt.genes$type == "tumor"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPMnorm)
  M <- assays(setmp)$logCPMnorm[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

<br/>   

###Batch identification

Now we're going to analyze our data in order to search for batch effect that could interfere with the biological signal. First, we analyze some of the information contained in the barcode, such as tissue source site, center of sequenciation, plate and portion and analyte combinations. We use two approaches to try to identify batch effect, the Hierarchical Clustering and a Multidimensional Scaling plot.

```{r}
tss <- substr(colnames(se.filt.genes), 6, 7)
table(tss)
center <- substr(colnames(se.filt.genes), 27, 28)
table(center)
plate <- substr(colnames(se.filt.genes), 22, 25)
table(plate)
portionanalyte <- substr(colnames(se.filt.genes), 18, 20)
table(portionanalyte)
samplevial <- substr(colnames(se.filt.genes), 14, 16)
table(samplevial)
```

<br/>  


####TSS batch effect

```{r}
table(data.frame(TYPE=se.filt.genes$type, TSS=tss))
```
```{r sampleClusteringTSS, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S6: Hierarchical clustering of the samples."}
#logCPM <- cpm(dge.filt, log=TRUE, prior.count=3)
d <- as.dist(1-cor(assays(se.filt.genes)$logCPMnorm, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(tss))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(se.filt.genes)
outcome <- paste(substr(colnames(se.filt.genes), 9, 12), as.character(se.filt.genes$type), sep="-")
names(outcome) <- colnames(se.filt.genes)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("TSS", sort(unique(batch)), levels(factor(tss))), fill=sort(unique(batch)))

```

```{r mdsTSS, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S7: Multidimensional scaling plot of the samples."}
plotMDS(dge.filt, labels=outcome, col=batch)
legend("bottomleft", paste("TSS", sort(unique(batch)), levels(factor(tss))),
       fill=sort(unique(batch)), inset=0.05)
```

<br/>  


####PLATE batch effect

```{r}
table(data.frame(TYPE=se.filt.genes$type, PLATE=plate))
```

```{r sampleClusteringPLATE, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S6: Hierarchical clustering of the samples."}
#logCPM <- cpm(dge.filt, log=TRUE, prior.count=3)
d <- as.dist(1-cor(assays(se.filt.genes)$logCPMnorm, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(plate))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(se.filt.genes)
outcome <- paste(substr(colnames(se.filt.genes), 9, 12), as.character(se.filt.genes$type), sep="-")
names(outcome) <- colnames(se.filt.genes)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("PLATE", sort(unique(batch)), levels(factor(plate))), fill=sort(unique(batch)))

```

```{r mdsPLATE, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S7: Multidimensional scaling plot of the samples."}
plotMDS(dge.filt, labels=outcome, col=batch)
legend("bottomleft", paste("plate", sort(unique(batch)), levels(factor(plate))),
       fill=sort(unique(batch)), inset=0.05)
```

<br/>  


####PORTION-ANALYTE batch effect

```{r}
table(data.frame(TYPE=se.filt.genes$type, P.analyte = portionanalyte))
```
```{r sampleClusteringPortionanalyte, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S6: Hierarchical clustering of the samples."}
#logCPM <- cpm(dge.filt, log=TRUE, prior.count=3)
d <- as.dist(1-cor(assays(se.filt.genes)$logCPMnorm, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(portionanalyte))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(se.filt.genes)
outcome <- paste(substr(colnames(se.filt.genes), 9, 12), as.character(se.filt.genes$type), sep="-")
names(outcome) <- colnames(se.filt.genes)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("Portionanalyte", sort(unique(batch)), levels(factor(portionanalyte))), fill=sort(unique(batch)))

```

```{r mdsPortionanalyte, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S7: Multidimensional scaling plot of the samples."}
plotMDS(dge.filt, labels=outcome, col=batch)
legend("bottomleft", paste("Portionanalyte", sort(unique(batch)), levels(factor(portionanalyte))),
       fill=sort(unique(batch)), inset=0.05)

```

<br/>  

The Hierarchical clustering and the Multidimensional Scaling plots show a clear differentiation between normal samples and tumor samples. All four plots from the different elements of the barcode have a similar clustering and distribution. There is no clear effect from the batch indicators in the clustering, and it seems to have a balanced design (more clearly in tumor than in normal), although it seems that 4 tumor samples, 4770, 4698, 6087 and 4983 are clustered apart from the other samples in the hierarchical clustering, so we should consider taking apart this samples.


<br/>   

###Surrogate variable analysis

In order to clearly identify which values are affecting our outcome, we use the Surrogate Variable Analysis (SVA) package to find those surrogate variables that are interfering in the biological variability, and adjust for this non-biological (batch effect) variability.
```{r}
library(sva)
mod <- model.matrix(~ se.filt.genes$type, colData(se.filt.genes))
modzero <- model.matrix(~ 1, colData(se.filt.genes))
pv <- f.pvalue(assays(se.filt.genes)$logCPMnorm, mod, modzero)
sum(p.adjust(pv, method="fdr") < 0.01)
hist(pv,main="",las=1)
```

<br/>  

This is the p-value distribution of our set of genes. It shows a clear peak at low p-values, refered to those genes that are differently expressed between normal and tumor types of samples. We have a total of 7,578 differently expressed genes. Now, let's perform an SVA and adjust for the non-biological variability

```{r}
sv <- sva(assays(se.filt.genes)$logCPMnorm, mod, modzero)
sv$n
modsv <- cbind(mod, sv$sv)
modzerosv <- cbind(modzero, sv$sv)
pvsv <- f.pvalue(assays(se.filt.genes)$logCPMnorm, modsv, modzerosv)
sum(p.adjust(pvsv, method="fdr") < 0.01)
hist(pvsv,main="",las=1)
```

<br/>  

The SVA has found 27 surrogate variables, and adjusting for all of them increases the differentialy expressed genes to a number of 8796 genes.

<br/>

## CONCLUSIONS
After the analysis of the RNA-seq data of Kidney clear-cell-carcinoma, we have found 8,796 differentialy expressed genes between normal and tumor samples out of the 11,407 genes. 

We didn't find any significant batch effect out of the analysis of the four feature present in the sample barcode (tss, plate, center and portion-analyte). 

The hierarchical clustering showed four samples that clustered apart from the other samples, for this reason, they could be removed from the set.

We also performed a surrogate variable analysis to exclude any non-biological variation. We found 27 surrogate variables. 
