\documentclass[9pt,twocolumn,twoside]{gsajnl}
% Use the documentclass option 'lineno' to view line numbers

\articletype{inv} % article type
% {inv} Investigation 
% {gs} Genomic Selection
% {goi} Genetics of Immunity 
% {gos} Genetics of Sex 
% {mp} Multiparental Populations

\title{Kidney Renal Clear Cell Carcinoma expression analysis}

\author[$\ast$1]{Bofill A,}
\author[$\ast$1]{Castillo S,}
\author[$\ast$1]{Pérez A}


\affil[$\ast$]{Msc in Bioinformatics for Health Sciences, Pompeu Fabra University}

\keywords{Kidney; Carcinoma; Bioconductor; ASS. Sergio Castillo ...}

\runningtitle{GENETICS Journal Template on Overleaf} % For use in the footer 

\correspondingauthor{Corresponding Author}

\begin{abstract}
The abstract should be written for people who may not read the entire paper, so it must stand on its own. The impression it makes usually determines whether the reader will go on to read the article, so the abstract must be engaging, clear, and concise. In addition, the abstract may be the only part of the article that is indexed in databases, so it must accurately reflect the content of the article. A well-written abstract is the  most effective way to reach intended readers, leading to more robust search, retrieval, and usage of the article. 

\end{abstract}
\setboolean{displaycopyright}{true}
\begin{document}
\maketitle
\thispagestyle{firststyle}
\marginmark
\firstpagefootnote
\correspondingauthoraffiliation{Msc in Bioinformatics for Health Sciences, Pompeu Fabra University. adria.perez06@estudiant.upf.edu}
\vspace{-11pt}%



\section*{Introduction}

Kidney Renal Clear Cell Carcinoma (KIRC) is the most common type of kidney cancer (95\%)\emph{[cita cancer SEEE]}.  An estimated 62,700 new cases of kidney cancer are expected to be diagnosed in 2016, with 14.240 expected deaths (2,4\% of all death cancers).   The  5-year  and  10-year  relative  survival  rates  for  kidney  
cancer are 74\% and 62\% respectively.  Two-thirds of cases 
 are diagnosed at a local stage, for which the 5-year relative 
survival  rate  is  92\%, but when the tumour has spread from the kidney to other parts of the body, the rate lowers at 11\%\emph{[citar cancer SEEE?]}. 

The renal clear cell carcinoma is a malignant cancer from the renal parenchyma, originated in the tubules. The clear cells carcinoma is one of the four major histologic subtype, and the most common (75\%). Clear cells are a specific cell-type defined by a clear cytoplasm due to a high lipid content. This subtype is the least likely to reproduce, but has a strong resistance to chemotherapy and radiotherapy. The primary treatment is nephrectomy or partial nephrectomy, and sometimes laparoscopic techniques are used. 
	

Although KIRC is a common cancer, little has been done to its molecular characterization. The reason behind its high resistance to chemotherapy and radiotherapy is still unkown. Some molecular alterations have been identified. The PI3K/AKT pathway seems to be recurrently mutated, and a widespread DNA hypomethylation was associated with a mutation of the \textit{SETD2} methyltransferase. In aggressive cancers, there's evidence of a methabolic shift, with downregulation of the oxidative phosphorylation  and the fatty-acid degradation pathways and upregulation of the pentose phosphate pathway\emph{[comprehensive KIRC cit]}.

A molecular characterization of the KIRC is needed in order to identify the key molecular mechanisms that cause this cancer and target them in novel treatments. In this study,  a differential expression analysis of the RNA-Seq data provided by the TCGA has been performed, with a further functional enrichment analysis of GO terms and a Gene Set Enrichment Analysis to identify the molecular profile of this cancer.

\section*{Materials and Methods}

\subsection*{Data characterization}
The kidney Renal Clear-Cell Carcinoma RNA-seq data used in this study has been extracted from The Cancer Genome Atlas \href{http://www.cancergenome.nih.gov}{TCGA}. The initial data available from TCGA dataset comprised 542 tumours and 72 normal samples. The tumour group include 187 females and 337 males, while the normal one is formed by 20 female and 52 males. Analysing this data samples, 38 paired samples were found. Each one of the samples have a total of 20115 genes. 

\subsection*{Statistics and data analysis}
All the analyses have been performed with R. You could find more information about that in: \url{www.r-project.org}. CITAAAAAA!!! EdgeR package from Bioconductor has been used for the differentially expressed gene identification analysis.  CITAAAAAA!

\subsubsection*{Quality and normalization }
: The library size of the samples has been analysed and major differences in sequencing depth were found. We filtered out the samples with a low sequencing depth (< 45 Millions per read). We obtained a filtered dataset of 298 tumour and 48 normal samples. To obtain a more accurate information and to reduce the size of our set, we determined to use a paired design, taking into account the patient variability. For this reason, we only considered the 38 paired samples for the further analyses.

To filter out the genes with a low or without expression, we analysed the distribution of expression levels among genes using the log Counts Per Millions measure (logCPM) (CITAAAAAAA). (GRAFICA? S7). We determined a cut-off of 1 logCPM. With this filter we reduce the number of genes from 20,115 to 12,495 genes.

The Trimmed Mean of M.-values method (TMM) (CITAAAA! Robinson and Oshlack, 2010), implemented in the EdgeR package, was used to normalize the expression values.

Analysing MA-plots, a normal sample (TGCA-CW-5591) was found to have an abnormal major gene expression bias in the association between the fold-changes and average expression, resulting in a large dependence between both variables of the MA plot. For this reason, this sample and its paired in the tumour set were removed from the analysis.


\subsubsection*{Batch Effect Analyses}
: Four possible sources of variation were considered for the batch effect identification: Gender, Tissue Source Site (TSS), Plate, and portion analyte. For each one of these batch indicators, we performed a hierarchical clustering using the Spearman correlation coefficient and a multidimensional scaling plot, in order to assess their possible effect. We conclude that none of them confound the primary source of variation, which in this case is the cell-type (tumour/normal).

\subsubsection*{Differential expression analysis	}
: To perform the differentially expressed gene identification analysis, we generate a two factors linear model, taking into account the cell-type variable (normal/tumour type) and the patient.

We modelled the mean variance trend of logCPM values, computing the weights of this relationship at the individual observation level. In order to perform this part, we applied the voom function, implemented in the limma R-package, which is part of the BioConductor project (CITAAAAA!). A Surrogate Variable Analysis (SVA) was performed to identify the possibles sources of variation that are not related with our variables of interest (cell-type). (Leek and Storey 2007).

After fitting the data to the linear model, we applied the 	empirical Bayes approach (eBayes) that should result in a far more stable inference (CITAAA: Smyth GK 2004 linear models and empirical baytes...). A False Discovery Rate (FDR)(CITAAAAA: Benjamini, Y. and Y. Hochberg, 1995 Controlling the False Dis- covery Rate: A Practical and Powerful Approach to Multi- ple Testing. Journal of the Royal Statistical Society. Series B (Methodological) 57: 289 – 300.) cut-off of 0.01 was applied to classify the genes as over-expressed, under-expressed or without changed in expression. In order to reduce the number of differentially expressed genes, we classified the genes in two further groups: strongly over-expressed, using a log Fold-changes cut-off of 5 (logFC > 5), and strongly under-expressed (logFC < 5). 

\subsubsection*{Gene Set Enrichment Analysis}
:  Using GSEABase package from the BioConductor project, the simple Gene Set Enrichment Analysis (GSEA) (CITAAA) algorithm was applied in order to assess differences in  expression at the pathway level. We used the data set called c2BroadSets from the GSVAdata to obtain the different gene sets, restricting the pathways to the ones from KEGG, REACTOME and BIOCARTA. (CITAAAAAAA).


\subsubsection*{Functional Enrichment}
: A Gene Ontology analysis (GO) was performed with the differentially expressed genes set. Fisher's exact test was applied to obtain the most representative biological process gene ontology terms (P< 0.01). Three enrichment tests were performed in three different gene sets: the whole list of differentially expressed genes, the over-expressed gene set and the under-expressed gene set. The GO results are available in the supporting materials together with the raw p-values.

\subsubsection*{Hierarchical Clustering using strongly DE Genes}
: We use the strongly over-expressed genes and under-expressed genes, a total of 77 genes, to determine if these genes alone are able to separate our samples in two clusters using a Hierarchical Clustering approach (CITA SPEARRMAAAN). This approach has been applied over the 38 paired samples and also over the whole set of samples (614 samples).

\subsection*{Data Availability}

At the end of the Materials and Methods section, include a statement on reagent and data availability. Please read the Data and Reagent Policy before writing the statement. Make sure to list the accession numbers or DOIs of any data you have placed in public repositories. List the file names and descriptions of any data you will upload as supplemental information. The statement should also include any applicable IRB numbers. You may include specifications for how to properly acknowledge or cite the data.

For example: Strains are available upon request. File S1 contains detailed descriptions of all supplemental files. File S2 contains SNP ID numbers and locations. File S3 contains genotypes for each individual. Sequence data are available at GenBank and the accession numbers are listed in File S3. Gene expression data are available at GEO with the accession number: GDS1234. Code used to generate the simulated data is provided in file S4. 


\section*{Results and Discussion}

The results and discussion should not be repetitive. The results section should give a factual presentation of the data and all tables and figures should be referenced; the discussion should not summarize the results but provide an interpretation of the results, and should clearly delineate between the findings of the particular study and the possible impact of those findings in a larger context. Authors are encouraged to cite recent work relevant to their interpretations. Present and discuss results only once, not in both the Results and Discussion sections. It is sometimes acceptable to combine results and discussion. The text should be as succinct as possible. Heed Strunk and White's dictum: "Omit needless words!"
\newpage

\section*{Additional guidelines}

\subsection*{Numbers} In the text, write out numbers nine or less except as part of a date, a fraction or decimal, a percentage, or a unit of measurement. Use Arabic numbers for those larger than nine, except as the first word of a sentence; however, try to avoid starting a sentence with such a number.

\subsection*{Units} Use abbreviations of the customary units of measurement only when they are preceded by a number: "3 min" but "several minutes". Write "percent" as one word, except when used with a number: "several percent" but "75\%." To indicate temperature in centigrade, use ° (for example, 37°); include a letter after the degree symbol only when some other scale is intended (for example, 45°K).

\subsection*{Nomenclature and Italicization} Italicize names of organisms even when  when the species is not indicated.  Italicize the first three letters of the names of restriction enzyme cleavage sites, as in HindIII. Write the names of strains in roman except when incorporating specific genotypic designations. Italicize genotype names and symbols, including all components of alleles, but not when the name of a gene is the same as the name of an enzyme. Do not use "+" to indicate wild type. Carefully distinguish between genotype (italicized) and phenotype (not italicized) in both the writing and the symbolism.

\section*{In-text Citations}

Add citations using the \verb|\citep{}| command, for example \citep{neher2013genealogies} or for multiple citations, \citep{neher2013genealogies, rodelsperger2014characterization}

\section*{Examples of Article Components}
\label{sec:examples}

The sections below show examples of different header levels, which you can use in the primary sections of the manuscript (Results, Discussion, etc.) to organize your content.

\section*{First level section header}

Use this level to group two or more closely related headings in a long article.

\subsection*{Second level section header}

Second level section text.

\subsubsection*{Third level section header:}

Third level section text. These headings may be numbered, but only when the numbers must be cited in the text. 

\section*{Figures and Tables}

Figures and Tables should be labelled and referenced in the standard way using the \verb|\label{}| and \verb|\ref{}| commands.

\subsection*{Sample Figure}

Figure \ref{fig:spectrum} shows an example figure.

\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{example-figure.png}
\caption{Example figure from \url{10.1534/genetics.114.173807}. Please include your figures in the manuscript for the review process. You can upload figures to Overleaf via the Project menu. Upon acceptance, we'll ask for your figure files to be uploaded in any of the following formats: TIFF (.tiff), JPEG (.jpg), Microsoft PowerPoint (.ppt), EPS (.eps), or Adobe Illustrator (.ai).  Images should be a minimum of 300 dpi in resolution and 500 dpi minimum if line art images.  RGB, CMYK, and Grayscale are all acceptable. Halftones should be high contrast with sharp detail, because some loss of detail and contrast is inevitable in the production process. Figures should be 10-20 cm in width and 1-25 cm in height. Graph axes must be exactly perpendicular and all lines of equal density.
Label multiple figure parts with A, B, etc. in bolded type, and use Arrows and numbers to draw attention to areas you want to highlight. Legends should start with a brief title and should be a self-contained description of the content of the figure that provides enough detail to fully understand the data presented. All conventional symbols used to indicate figure data points are available for typesetting; unconventional symbols should not be used. Italicize all mathematical variables (both in the figure legend and figure) , genotypes, and additional symbols that are normally italicized.  
}%
\label{fig:spectrum}
\end{figure}

\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{example-figure}
\caption{Example movie (the figure file above is used as a placeholder for this example). \textit{GENETICS} supports video and movie files that can be linked from any portion of the article - including the abstract. Acceptable formats include .asf, avi, .wav, and all types of Windows Media files.   
}%
\end{figure}


\subsection*{\url Table}

Table \ref{tab:shape-functions} shows an example table. Avoid shading, color type, line drawings, graphics, or other illustrations within tables. Use tables for data only; present drawings, graphics, and illustrations as separate figures. Histograms should not be used to present data that can be captured easily in text or small tables, as they take up much more space.  

Tables numbers are given in Arabic numerals. Tables should not be numbered 1A, 1B, etc., but if necessary, interior parts of the table can be labeled A, B, etc. for easy reference in the text.  


Leek, J. T. and J. D. Storey, 2007 Capturing heterogeneity in gene expression studies by surrogate variable analysis. PLoS Genetics 3: 1724–1735.

\begin{table*}[htbp]
\centering
\caption{\bf Students and their grades}
\begin{tableminipage}{\textwidth}
\begin{tabularx}{\textwidth}{XXXX}
\hline
Student & Grade\footnote{This is an example of a footnote in a table. Lowercase, superscript italic letters (a, b, c, etc.) are used by default. You can also use *, **, and *** to indicate conventional levels of statistical significance, explained below the table.} & Rank & Notes \\
\hline
Alice & 82\% & 1 & Performed very well.\\
Bob & 65\% & 3 & Not up to his usual standard.\\
Charlie & 73\% & 2 & A good attempt.\\
\hline
\end{tabularx}
  \label{tab:shape-functions}
\end{tableminipage}
\end{table*}


\bibliography{example-bibliography}

\end{document}