\documentclass[9pt,twocolumn,twoside]{gsajnl}
% Use the documentclass option 'lineno' to view line numbers

\articletype{inv} % article type
% {inv} Investigation 
% {gs} Genomic Selection
% {goi} Genetics of Immunity 
% {gos} Genetics of Sex 
% {mp} Multiparental Populations

\title{Kidney Renal Clear Cell Carcinoma expression analysis}

\author[$\ast$1]{Bofill A,}
\author[$\ast$1]{Castillo S,}
\author[$\ast$1]{Pérez A}


\affil[$\ast$]{Msc in Bioinformatics for Health Sciences, Pompeu Fabra University}

\keywords{Kidney; Carcinoma; Bioconductor}

\runningtitle{GENETICS Journal Template on Overleaf} % For use in the footer 

\correspondingauthor{Corresponding Author}

\begin{abstract}

Renal Clear Cell Carcinoma is the most common type of Kidney cancer, but very few things are known about his molecular insights. In this  study, we perform a differential expression analysis coupled with a function enrichment analysis and a gene set enrichment analysis in order to identify the genes and paths that characterize  the molecular functioning of this type of cancer. We also build a Naive Bayes classifier to decide whether a sample is from a tumour or a normal tissue by their expression data. The results obtained indicate that renal clear cell carcinomas suffer a metabolic shift towards a strong fatty-acid synthesis by downregulating their degradation and the oxidative phosphorilation. Results also single out PD-1 as a key factor for the immune system evasion of the tumour.


\end{abstract}
\setboolean{displaycopyright}{true}
\begin{document}
\maketitle
\thispagestyle{firststyle}
\marginmark
\firstpagefootnote
\correspondingauthoraffiliation{Msc in Bioinformatics for Health Sciences, Pompeu Fabra University. adria.perez06@estudiant.upf.edu}
\vspace{-11pt}%


\section*{Introduction}

Kidney Renal Clear Cell Carcinoma (KIRC) is the most common type of kidney cancer (95\%)\emph{[cita cancer SEEE]}.  An estimated 62,700 new cases of kidney cancer are expected to be diagnosed in 2016, with 14.240 expected deaths (2,4\% of all death cancers).   The  5-year  and  10-year  relative  survival  rates  for  kidney  
cancer are 74\% and 62\% respectively.  Two-thirds of cases 
 are diagnosed at a local stage, for which the 5-year relative 
survival  rate  is  92\%, but when the tumour has spread from the kidney to other parts of the body, the rate lowers at 11\%\emph{[citar cancer SEEE?]}. 

The renal clear cell carcinoma is a malignant cancer from the renal parenchyma, originated in the tubules. The clear cell carcinoma is one of the four major histologic subtypes, and the most common one (75\%). Clear cells are a specific cell-type defined by a clear cytoplasm due to a high lipid content. This subtype is the least likely to reproduce, but has a strong resistance to chemotherapy and radiotherapy. The primary treatment is nephrectomy or partial nephrectomy, and sometimes laparoscopic techniques are used. 
	

Although KIRC is a common cancer, little has been done to its molecular characterization. The reason behind its high resistance to chemotherapy and radiotherapy is still unkown. Some molecular alterations have been identified. The PI3K/AKT pathway seems to be recurrently mutated, and a widespread DNA hypomethylation was associated with a mutation of the \textit{SETD2} methyltransferase. In some cancers, there's evidence of a methabolic shift, with downregulation of the oxidative phosphorylation  and the fatty-acid degradation pathways and upregulation of the pentose phosphate pathway\emph{[Comprehensive guide to clear cell renal cell carcinoma]}.

A molecular characterization of the KIRC is needed in order to identify the key molecular mechanisms that cause this cancer and target them in novel treatments. In this study,  a differential expression analysis of the RNA-Seq data provided by the TCGA has been performed, with a further functional enrichment analysis of GO/KEGG terms and a Gene Set Enrichment Analysis (GSEA) to identify the molecular profile of this cancer.

\section*{Materials and Methods}
\subsection*{Data characterization}
The Kidney Renal Clear-Cell Carcinoma RNA-seq data used in this study has been extracted from The Cancer Genome Atlas (\href{http://www.cancergenome.nih.gov}{TCGA}). The initial data available from TCGA dataset comprised 542 tumours and 72 normal samples. The tumour group include 187 females and 337 males, while the normal one is formed by 20 female and 52 males. Analysing this data samples, 38 paired samples were found. Each sample has a total of 20115 genes. 

\subsection*{Statistics and data analysis}
All the analyses have been performed with R. You could find more information about that in: \url{www.r-project.org}. \href{https://bioconductor.org/packages/release/bioc/html/edgeR.html}{EdgeR} \citep{Robinson2010} package from \href{https://bioconductor.org}{Bioconductor} \citep{Gentleman2004} has been used for the differentially expressed gene identification analysis.  CITAAAAAA!

\subsubsection*{Quality and normalization }
: The library size of the samples has been analysed and major differences in sequencing depth were found. We filtered out the samples with a low sequencing depth (< 45 Millions per read). We obtained a filtered dataset of 298 tumour and 48 normal samples. To obtain a more accurate information and to reduce the size of our set, we determined to use a paired design, taking into account the patient variability. For this reason, we only considered the 38 paired samples for the further analyses.

To filter out the genes with a low or without expression, we analysed the distribution of expression levels among genes using the log Counts Per Millions measure (logCPM) (CITAAAAAAA). (GRAFICA? S7). We determined a cut-off of 1 logCPM. With this filter we reduce the number of genes from 20,115 to 12,495 genes.

The Trimmed Mean of M.-values method (TMM) (CITAAAA! Robinson and Oshlack, 2010), implemented in the EdgeR package, was used to normalize the expression values.

Analysing MA-plots, a normal sample (TGCA-CW-5591) was found to have an abnormal major gene expression bias in the association between the fold-changes and average expression, resulting in a large dependence between both variables of the MA plot. For this reason, this sample and its paired in the tumour set were removed from the analysis.


\subsubsection*{Batch Effect Analyses}
: Four possible sources of variation were considered for the batch effect identification: Gender, Tissue Source Site (TSS), Plate, and portion analyte. For each one of these batch indicators, we performed a hierarchical clustering using the Spearman correlation coefficient and a multidimensional scaling plot, in order to assess their possible effect. We conclude that none of them confound the primary source of variation, which in this case is the cell-type (tumour/normal).

\subsubsection*{Differential expression analysis	}
: To perform the differentially expressed gene identification analysis, we generate a two factors linear model, taking into account the cell-type variable (normal/tumour type) and the patient.

We modelled the mean variance trend of logCPM values, computing the weights of this relationship at the individual observation level. In order to perform this part, we applied the voom function, implemented in the limma R-package, which is part of the BioConductor project (CITAAAAA!). A Surrogate Variable Analysis (SVA) was performed to identify the possibles sources of variation that are not related with our variables of interest (cell-type). (Leek and Storey 2007).

After fitting the data to the linear model, we applied the 	empirical Bayes approach (eBayes) that should result in a far more stable inference (CITAAA: Smyth GK 2004 linear models and empirical baytes...). A False Discovery Rate (FDR)(CITAAAAA: Benjamini, Y. and Y. Hochberg, 1995 Controlling the False Dis- covery Rate: A Practical and Powerful Approach to Multi- ple Testing. Journal of the Royal Statistical Society. Series B (Methodological) 57: 289 – 300.) cut-off of 0.01 was applied to classify the genes as over-expressed, under-expressed or without changed in expression. In order to reduce the number of differentially expressed genes, we classified the genes in two further groups: strongly over-expressed, using a log Fold-changes cut-off of 5 (logFC > 5), and strongly under-expressed (logFC < -5). 

\subsubsection*{Functional Enrichment}
: A Gene Ontology analysis (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) analysis was performed with the differentially expressed genes set. Fisher's exact test was applied to obtain the most representative biological process gene ontology terms or KEGG pathways (P< 0.01). The enrichment tests were performed in three different gene sets: the whole list of differentially expressed genes, the over-expressed gene set and the under-expressed gene set. The GO and KEGG results are available in the supporting materials together with the raw p-values.

\subsubsection*{Gene Set Enrichment Analysis}
:  Using GSEABase package from the BioConductor project, the Simple Gene Set Enrichment Analysis (GSEA) (CITAAA) algorithm was applied in order to assess differences in  expression at the pathway level. We used the data set called c2BroadSets from the GSVAdata to obtain the different gene sets, restricting the pathways to the ones from KEGG, REACTOME and BioCarta. (CITAAAAAAA). We used an FDR < 1\% to call the pathways differentially expressed.

\subsubsection*{Hierarchical Clustering and Naive Bayes classifier}
: We used the strongly over-expressed genes and under-expressed genes, a total of 199 genes, to determine if these genes alone are able to separate our samples in two clusters using a Hierarchical Clustering approach (CITA SPEARRMAAAN). This approach has been applied over the 38 paired samples and also over the whole set of samples (614 samples).

Applying the R package e1071, a Naive Bayes Classifier (CITAA) was generated, using the paired samples as the training set, considering only the 199 strongly DE genes as features. This classifier was used to predict the cell type of the remaining 540 samples. 

\subsection*{Data Availability}

At the end of the Materials and Methods section, include a statement on reagent and data availability. Please read the Data and Reagent Policy before writing the statement. Make sure to list the accession numbers or DOIs of any data you have placed in public repositories. List the file names and descriptions of any data you will upload as supplemental information. The statement should also include any applicable IRB numbers. You may include specifications for how to properly acknowledge or cite the data.

For example: Strains are available upon request. File S1 contains detailed descriptions of all supplemental files. File S2 contains SNP ID numbers and locations. File S3 contains genotypes for each individual. Sequence data are available at GenBank and the accession numbers are listed in File S3. Gene expression data are available at GEO with the accession number: GDS1234. Code used to generate the simulated data is provided in file S4. 


\section*{Results and Discussion}
\subsection*{Differentially Expressed Analysis}
The DE analysis shows 9,108 differentially expressed genes in KIRC tumour, being 4,014 under-expressed and 5,094 over-expressed. Out of these DE genes, 46 were classified as strongly over-expressed and 153 as strongly under-expressed. The volcano plot represented in Fig. 1 shows the different filters applied to the whole set of genes.

Analysing individual genes previously found to be mutated in the KIRC development, the chromatin regulators SETD2 and PBRM1 are over-expressed, even though they are tumour-supressor genes as they promote a correct chromatin remodelling and DNA repair [CITAR 2:Comprehensive molecular characterization of clear cell renal cell carcinoma / SETD2 loss-of-function promotes renal cancer branched evolution through replication stress and impaired DNA repair]. While PI3K/AKT pathway has been targeted as strong therapeutical target (CITAA: Comprehensive molecular characterization of clear cell renal cell carcinoma), we did not find any over-expressed gene of this pathway (PTEN/PI3K/PDK1/AKT/MTOR).

\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{figures/fig1_volcano.png}
\caption{Volcano plot of the differential expression analysis of Kidney Renal Clear Cell Carcinoma. The horizontal dashed line represents the FDR cutoff of 0.01 used for calling DE genes. After calling 9,108 genes as DE, we reduced the number of genes using a Fold Change cutoff of 5 and -5 (vertical lines), resulting in two groups of stronly over-expressed and strongly under-expressed genes (in yellow). }%
\label{fig:spectrum}
\end{figure}

\subsection*{Functional Enrichment}
The Functional Enrichment analysis using GO terms revealed 96 over-represented terms and 59 under-represented [Full results here and here]. The most representative terms in the over-represented group include cell-division and proliferation, intrinsic properties in any malignant cell. 

Another group of representative terms are the ones related with regulation of immune response and T-cell functioning. This suggests that KIRC may cause a huge deregulation of the local immune response to the tumour. 

When looking at the under-represented terms, many of them are related with renal function like ion transport and small molecule metabolism. Several terms related with the oxidative phosphorilation pathway are under-represented along with fatty-acid degradation. This metabolical shift to fatty-acid synthesis and inactivation of oxidative phosphorilation was previously indentified in KIRC [CITAAA: Comprehensive molecular characterization of clear cell renal cell carcinoma].

KEGG analysis results show that among the most represented pathways over the over-expressed DE genes, some KEGG pathways terms related to the immune system response, such as primary immunodeficiency and T-cell receptor signaling pathway, can be found.  These results are in line with the suggested hypothesis of an immune system deregulation in KIRC. Furthermore, the p53 signaling pathway is also found as one of the most over-represented pathways.

On the other hand, oxidative phosphorylation and fatty acid degradation are found to be among the most represented pathways over the under expressed DE genes. As seen with the GO terms, this indicates a metabolic shift in KIRC to promote fatty acid synthesis and inactivates the oxidative phosphorylation.

\subsection*{Gene Set Enrichment Analysis}
The GSEA results show 677 DE pathways. In this study we only focused on the top ones. This results can be found on supplementary materials (Fig. S22).


The analysis highlights again the loss of kidney function and the metabolic shift to the fatty acid synthesis  (Fig. 3A). We found several pathways of small molecules, cation and glucose transport across the membrane that are under-expressed in KIRC, suggesting that the renal function is disrupted. As seen before in the Functional Enrichment analysis results, we found the oxidative phosphorilation and the fatty acid degradation pathways under-expressed, indicating an increase of fatty acid synthesis and  an inactivation of the TCA cylce in favour of the pentose phosphate pathway, as suggested in [cita: Comprehensive guide to clear cell renal cell carcinoma]. 

As previously mentioned in the Functional Enrichment results, many immune system sets are found to be differentially expressed. Primary immunodeficiency, T-cell apopotosis and PD1 signalling pathways, show us clues of how KIRC evades the immune function of the organism. The BioCarta T-cell apoptosis pathway is characteristic of the HIV infection and describes how the virus induces T-cell apoptosis. The fact that KIRC presents this feature indicates that this may be an useful mechanism for this tumour to not be recognized as a malignant cell.

Programmed Cell Death Protein 1 (PD-1/CD279) is a cell surface receptor that belongs to the immunoglobulin super-family and is expressed in T-cells. It functions as an immune checkpoint, negatively regulating the immune response (CITA Loise M Francisco...). PD1 plays an important role in down-regulating the immune system, reducing autoimmunity and promoting self-tolerance through the induction of antigen-specific T-cell apoptosis.
PD-1 has been involved in the tumour cell evasion of the host immune system (CITA Yoshiko iwai). This argument is in line with the over-expression of PD-1 found in our GSEA results. This protein has been proposed as target for several drugs that are currently in different trial stages, such as Nivolumab, used as a possible second line treatment for renal cell carcinoma \cite{nivolumab}.


\subsection*{Naive Bayes classifier}
With a Hierarchical Clustering approach, we can observe if the strongly DE genes separate the samples in different clusters. We applied this method to the paired samples and we can observe a clear separation between normal and tumour samples in two clusters (Fig. 2A). These results show that these 199 strongly DE genes could be used for building a classification method based on machine learning (CTAAA). 

\section*{Conclusions}

Performing a general DE analysis coupled with functional enrichment and gene-set enrichment analysis has provided us with a decent characterisation of the renal clear cell carcinoma. We have been able to single out the differentially expressed genes in this type of cancer, and analyse this data to pinpoint the main molecular pathways that lead to cell malignancy in KIRC. 

The analysis has identyfied the main metabolic route of the tumour, promoting the fatty-acid synthesis and inhibiting the oxidative phosphorilation, validating the histological feature of a clear cytoplasm in clear cells. It has also spotted PD-1 signaling pathway as a major tool to evade the immune system, and
stood out a general immunitary evasion and downregulation from the tumour.

This results help us to establish putative targets for pharmacological treatment, in order to improve the current treatments. PD-1 has been identified as an important target for cancer immunotherapy, activating the immune system to attack malignant cells, and the PD-1 inhibitor Nivolumab[cite]has been used to treat renal cell carcinoma.


This demonstrates how our analysis was able to identify a target protein that has drug inhibitors already in clinical trials, reinforcing our  methodology as a system to molecularly characterize tumours and point out putative targets to adress cancer treatments.

\begin{figure*}[htbp]
\centering
\includegraphics[width=\textwidth]{figures/fig2.png}
\caption{Hierarchical clustering of paired samples using the 199 differentially expressed genes in KIRC. After calling the DE genes using an FDR of 1\%, all the samples used for this expression study were clustered by means of the Spearman's correlation coefficient, but reducing the feature vector to only the logCPM of the differentially expressed genes. }%
\label{fig:spectrum}
\end{figure*}

The application of this approach to the whole set of samples results in a more complex set of clusters. As we can see in Fig. 2B, although almost the entire set of normal and tumour samples, cluster together, some tumour samples show a unexpected clustering behaviour. These samples are closer to the cluster of normal samples, leading us to determine that these strongly DE genes do not capture the whole variability between tumour and normal samples.

With the Naive Bayes Classifier, over the prediction of the cell type of the unpaired samples, we computed the specificity and sensitivity and we obtain a value of 0.971 and 0.921 respectively. The F-measure computed with these values results in a value of 0.0609. In the x CITAAAAA article, a similar classifier was built, but using a Support Vector Machine instead of a Naive Bayer Classifier. Although K-Fold Cross Validation was not done in our study, which could explain the differences in the evaluation of the effectiveness of this classifier, the specificity and sensitivity results are very similar.


\section*{In-text Citations}

Add citations using the \verb|\citep{}| command, for example \citep{neher2013genealogies} or for multiple citations, \citep{neher2013genealogies, rodelsperger2014characterization}

\section*{Examples of Article Components}


\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.5\hsize}X}
\begin{table*}[htbp]
\centering
\caption{\bf Five most over-represented Biological Process GO terms in over-expressed genes in KIRC. }
\begin{tableminipage}{\textwidth}
\begin{tabularx}{\textwidth}{ssssssb}
\hline
\footnote{This is an example of a footnote in a table. Lowercase, superscript italic letters (a, b, c, etc.) are used by default. You can also use *, **, and *** to indicate conventional levels of statistical significance, explained below the table.}
GOBPID & Pvalue & OddsRatio & ExpCount & Count & Size & Term\\
\hline
GO:0051301 & 0.000 & 1.534 & 165 & 206 & 402 & cell division\\
GO:0051251 & 0.000 & 2.064 & 54 & 77 & 131 & positive regulation of lymphocyte activation\\
GO:0007156 & 0.000 & 2.469 & 35 & 53 & 84 & homophilic cell adhesion via plasma membrane adhesion molecules\\
GO:0008283 & 0.000 & 1.275 & 459 & 518 & 1117 & cell proliferation\\
GO:0032946 & 0.000 & 2.634 & 27 & 42 & 65 & positive regulation of mononuclear cell proliferation\\
\hline
\end{tabularx}
  \label{tab:shape-functions}
\end{tableminipage}
\end{table*}

Leek, J. T. and J. D. Storey, 2007 Capturing heterogeneity in gene expression studies by surrogate variable analysis. PLoS Genetics 3: 1724–1735.

\subsection*{Referennnnn}
\bibliographystyle{ieetr}
\bibliography{bibliography}

\end{document}